src/index.js

```diff
@@ -544,7 +544,7 @@ function Component(props, context, opts) {
 		newComponentHook.call(this, props, context);
 	}
 }
-extend(Component.prototype = new PreactComponent(), {
+extend(Component.prototype = PreactComponent.prototype, {
```

Comment:
instead `extend(Component.prototype, PreactComponent.prototype)` may be no mutate.

in this case, can't use extends mode(Object.create or new PreactComponent), because we need flat methods of prototype.